bindings/azure/cosmosdb/cosmosdb.go

```diff
@@ -46,11 +47,26 @@ func (c *CosmosDB) Init(metadata bindings.Metadata) error {
 	}
 
 	c.partitionKey = m.PartitionKey
-	client := documentdb.New(m.URL, &documentdb.Config{
-		MasterKey: &documentdb.Key{
+
+	// Create the client; first, try authenticating with a master key, if present
+	var config *documentdb.Config
+	if m.MasterKey != "" {
+		config = documentdb.NewConfig(&documentdb.Key{
 			Key: m.MasterKey,
-		},
-	})
+		})
+	} else {
+		// Fallback to using Azure AD
+		env, err := azure.NewEnvironmentSettings("cosmosdb", metadata.Properties) //nolint:shadow
```

Comment:
Thanks for explaining. I don't feel strongly about this, although I'm not sure what fix you have in mind that makes it awkward? I would just rename `err` to `envErr` and `sptErr` in their respective contexts to avoid the shadowing.

Reply:
Oh, that would have been the easy fix, wouldn't it? 😅
I was thinking to define `var env *type` (can't remember the type here) in the line before, so then there were no new variables to define during the instantiation on what is currently line 59. However doing that would have required adding another import too.
I can change to a different variable name.